Feature(authentication): add Linux support using zbus and polkit (#5)
project-robius/robius: crates/authentication/README.md

<uses-permission android:name="android.permission.USE_BIOMETRIC" />
```

## Usage on Linux

On Linux, `robius-authentication` uses **polkit** to request authorization via the
desktop environment's native authentication prompt (GNOME/KDE/etc).

> [!IMPORTANT]
> **Ensure a polkit agent is running**
>  
> The prompt is displayed by a polkit authentication agent (GNOME/KDE usually start one automatically).
> If no agent is running (headless/SSH), no prompt will appear and auth will fail.

### Add a polkit action policy file

#### Development & debugging

During development and debugging, you can simplify the process using the following steps:

1. Place the policy file within the project, for example in resources/com.yourapp.policy.

2. Install it once on your development machine using sudo:

```bash
sudo install -Dm644 resources/com.yourapp.policy \
  /usr/share/polkit-1/actions/com.yourapp.policy
```

3. Verify functionality with pkaction:

```bash
pkaction --action-id com.yourapp.authenticate --verbose
```

Log back into your desktop session (or restart polkitd), then run your example.

> Plicy file example:

Reviewer: ```suggestion
Policy file example:
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<policyconfig>
  <action id="com.yourapp.authenticate">
    <description>Authenticate to use YourApp</description>
    <message>Authentication is required</message>
    <defaults>
      <allow_active>auth_admin_keep</allow_active>
    </defaults>
  </action>
</policyconfig>
````
#### 

> [!TIP]

> > Automatic addition during runtime is not recommended and should generally be avoided.

Reviewer: what does "automatic addition" mean? Addition of what? please clarify.

The reason is not technical feasibility, but rather security/distribution policy restrictions:
> 
> polkit actions (.policy files) are system-level security policies. By design, they must be written by package managers or administrators during installation. Ordinary applications should not be allowed to modify system security configurations during runtime.
> 
> Writing to `/usr/share/...` during runtime requires root privileges; allowing an app to elevate privileges to modify policy files is flagged as a security red flag by many distributions.
>
> The only recommended “automatic method” is installation-time automation.
This means packaging the `.policy` file alongside your deb/rpm/aur/flatpak package during installation.

## Example

```rust
use robius_authentication::{
    AndroidText, BiometricStrength, Context, Policy, PolicyBuilder, Text, WindowsText,
};

> // Linux ignores policy options like biometrics/password (kept for parity).

Reviewer: ```suggestion
// Linux ignores policy options like biometrics/password,
// but it's best to keep them in for proper operation on other platforms.
```

<uses-permission android:name="android.permission.USE_BIOMETRIC" />
```

## Usage on Linux

On Linux, `robius-authentication` uses **polkit** to request authorization via the
desktop environment's native authentication prompt (GNOME/KDE/etc).

> [!IMPORTANT]
> **Ensure a polkit agent is running**
>
> The prompt is displayed by a polkit authentication agent (GNOME/KDE usually start one automatically).
> If no agent is running (headless/SSH), no prompt will appear and auth will fail.

### Write policy file.

> You can crate your own application's policy file, also can crate by template policy file.

Reviewer: ```suggestion
You can create your application's own policy file from scratch, or also create one from a template policy file.
```

> Template policy file see: [`./examples/org.robius.authentication.policy`](./examples/org.robius.authentication.policy)

Reviewer: ```suggestion
See here for an example template policy file: [`./examples/org.robius.authentication.policy`](./examples/org.robius.authentication.policy)
```

### Quick Test Mode ⚠️

> Add the policy file to actions by manually executing the following command:

Reviewer: "add ... to actions"

what is "actions" here? i think we need more details to specify what "actions" refers to here.

```bash
sudo install -Dm644 com.yourapp.policy /usr/share/polkit-1/actions/
```

> Then, ensure your policy file is correctly installed.

Reviewer: nit
```suggestion
Then, ensure your policy file was correctly installed.
```

```bash
pkaction --action-id <YOUR_POLICY_File_ACTION_ID>
```

> During the test mode, you don't need to worry about the location of the policy file; just ensure it installs correctly.
>

> > You can also store it in advance under the unified packaging configuration folder (like `packaging`)to facilitate automatic installation of the policy file during release mode when users perform installation.

Reviewer: what does "store it in advance" mean? As in, create it in advance and move/install it into the proper directory _before_ installing the application? 

I feel like this sentence doesn't really help anyone; what were you hoping to convey here? Is this for developers or for users? or perhaps for people who are packaging up an app?

Missing a space after "(like `packaging`)"
